src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2620,6 +2557,78 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
             }
 
             return true;
+
+
+        }
+
+        Stream GetRefPeStream(DiagnosticBag metadataDiagnostics, EmitStreamProvider metadataPEStreamProvider) 
+        {
+            if (metadataDiagnostics.HasAnyErrors())
+            {
+                return null;
+            }
+
+            var refPeStream = metadataPEStreamProvider.GetOrCreateStream(metadataDiagnostics);
+            Debug.Assert(refPeStream != null || metadataDiagnostics.HasAnyErrors());
+            return refPeStream;
+        }
+
+        Stream GetPortablePdbStream(DiagnosticBag metadataDiagnostics, EmitStreamProvider pdbStreamProvider)
+        {
+            if (metadataDiagnostics.HasAnyErrors())
+            {
+                return null;
+            }
+
+            var portablePdbStream = pdbStreamProvider.GetOrCreateStream(metadataDiagnostics);
+            Debug.Assert(portablePdbStream != null || metadataDiagnostics.HasAnyErrors());
+            return portablePdbStream;
+        }
+
+        (Stream peStream, Stream signingStream, Stream retStream) GetPeStream(DiagnosticBag metadataDiagnostics, EmitStreamProvider peStreamProvider, bool metadataOnly)
```

Comment:
>retStream [](start = 55, length = 9)

what's "retStream"?

Reply:
I've changed the name to `selectedStream` and have added a doc comment.